WordPress/src/main/java/org/wordpress/android/ui/stories/media/StoryMediaSaveUploadBridge.kt

```diff
@@ -135,44 +141,45 @@ class StoryMediaSaveUploadBridge @Inject constructor(
 
                     // here we change the ids on the actual StoryFrameItems, and also update the flattened / composed image
                     // urls with the new URLs which may have been replaced after image optimization
-                    for (story in storyRepositoryWrapper.getImmutableStories()) {
-                        // find the MediaModel for a given Uri from composedFrameFile
-                        for (frame in story.frames) {
-                            // if the old URI in frame.composedFrameFile exists as a key in the passed map, then update that
-                            // value with the new (probably optimized) URL and also keep track of the new id.
-                            val oldUri = Uri.fromFile(frame.composedFrameFile)
-                            val mediaModel = oldUriToMediaFiles.get(oldUri)
-                            mediaModel?.let {
-                                val oldTemporaryId = frame.id ?: ""
-                                frame.id = it.id.toString()
+                    // find the MediaModel for a given Uri from composedFrameFile
```

Comment:
Side effect of this refactor: we were able to ditch this nested for loop running through all the Stories the StoryRepository currently holds, given the frames collection for the specific Story being saved is now being passed by paramater.